oneandone/src/main/java/org/apache/jclouds/oneandone/rest/compute/function/SingleServerApplianceToImage.java

```diff
@@ -74,7 +74,11 @@ static boolean is64Bit(int architecture) {
       return tryFind(asList(OsFamily.values()), new Predicate<OsFamily>() {
          @Override
          public boolean apply(OsFamily input) {
-            return osFamily.toLowerCase().contains(input.value().toLowerCase());
+            if (osFamily != null) {
```

Comment:
`osFamily` is the input parameter of the function. There's no need to do the same redundant check in each iteration. Remove this and directly return absent at the beginning if the input parameter is null. Same for the other function.